Source/Task/AsyncQueue.cpp

```diff
@@ -245,21 +294,34 @@ class Queue
             isEmpty = m_queueHead.Flink == &m_queueHead;
             if (isEmpty)
             {
+#if HC_PLATFORM_IS_MICROSOFT
                 ResetEvent(m_event);
+#endif
             }
         }
 
-        if (isEmpty && m_dispatchMode == AsyncQueueDispatchMode_FixedThread &&
-            GetThreadId(m_apcThread) == GetCurrentThreadId())
+        if (isEmpty && m_dispatchMode == AsyncQueueDispatchMode_FixedThread)
         {
-            SleepEx(0, TRUE);
+#if HC_PLATFORM_IS_MICROSOFT
+            if (GetThreadId(m_apcThread) == GetCurrentThreadId())
+            {
+                SleepEx(0, TRUE);
+            }
+#else
+            assert(false);
+#endif
         }
     }
 
     bool Wait(uint32_t timeout)
     {
+#if HC_PLATFORM_IS_MICROSOFT
         uint32_t w = WaitForSingleObject(m_event, timeout);
         return (w == WAIT_OBJECT_0);
+#else
+        assert(false);
```

Comment:
I assume this is temporary -- I imagine that being able to wait on a call to come in will be pretty common, even for a manual queue.